src/inspector_agent.cc

```diff
@@ -89,18 +90,18 @@ void SendVersionResponse(inspector_socket_t* socket) {
   SendHttpResponse(socket, buffer, len);
 }
 
-void SendTargentsListResponse(inspector_socket_t* socket) {
+void SendTargentsListResponse(inspector_socket_t* socket, int port) {
   const char LIST_RESPONSE_TEMPLATE[] =
       "[ {"
       "  \"description\": \"node.js instance\","
       "  \"devtoolsFrontendUrl\": "
             "\"https://chrome-devtools-frontend.appspot.com/serve_file/"
-            "@4604d24a75168768584760ba56d175507941852f/inspector.html\","
+            "@%s/inspector.html?experiments=true&v8only=true&ws=localhost:%d%s\","
       "  \"faviconUrl\": \"https://nodejs.org/static/favicon.ico\","
       "  \"id\": \"%d\","
       "  \"title\": \"%s\","
       "  \"type\": \"node\","
-      "  \"webSocketDebuggerUrl\": \"ws://%s\""
+      "  \"webSocketDebuggerUrl\": \"ws://localhost:%d%s\""
```

Comment:
Nice one!

Indeed, it would be nice to get rid of the hard coded localhost and use the
value from addr()

I'm not sure if it's binding to 0.0.0.0 or the local IP (enables remoting
over the network?)


On Wed, Jun 8, 2016 at 1:16 PM Myles Borins <notifications@github.com>
wrote:

> In src/inspector_agent.cc
> <https://github.com/nodejs/node/pull/7232#discussion_r66331013>:
>
> >        "  \"faviconUrl\": \"https://nodejs.org/static/favicon.ico\","
> >        "  \"id\": \"%d\","
> >        "  \"title\": \"%s\","
> >        "  \"type\": \"node\","
> > -      "  \"webSocketDebuggerUrl\": \"ws://%s\""
> > +      "  \"webSocketDebuggerUrl\": \"ws://localhost:%d%s\""
>
> I'm not 100% that this should be localhost... it should potentially be the
> ip associated with the addr incase there is remote debugging going on. Open
> to modifying this but wanted to check first
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/nodejs/node/pull/7232/files/aca56db0c4691b6080290792ec8318c04dcb6347#r66331013>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAKl9ySsaar49QaCCK7P28BUgi8ZlXSJks5qJyMlgaJpZM4IxWgR>
> .
>